lib/libspl/timestamp.c

```diff
@@ -51,7 +53,7 @@ print_timestamp(uint_t timestamp_fmt)
 		fmt = nl_langinfo(_DATE_FMT);
 
 	if (timestamp_fmt == UDATE) {
-		(void) printf("%ld\n", t);
+		(void) printf("%" PRIu64 "\n", (uint64_t)t);
```

Comment:
Just for consistently with the rest of the code I'd suggest using a "%llu" here and casting it to a `u_longlong_t`.  This is how we've handled it elsewhere.